sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlIO.java

```diff
@@ -401,7 +415,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
 
     @VisibleForTesting
     BoundedSource<T> createSource() {
-      XmlSource<T> source = new XmlSource<>(this);
+      XmlSource<T> source = new XmlSource<>(this, getValidationEventHandler());
```

Comment:
Since XmlSource has visibility of "this", you shouldn't need to pass through the validation event handler to the XmlReader.

Within the XmlReader, you should be able to just invoke getCurrentSource().spec.getValidationEventHandler()